server/commitment_type.go

```diff
@@ -0,0 +1,107 @@
+package server
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net/http"
+
+	"github.com/Layr-Labs/eigenda-proxy/commitments"
+)
+
+type CommitmentType uint8
+
+const (
+	OptimismCommitmentType CommitmentType = iota
+	NakedCommitmentType
+	UnknownCommitmentType
+)
+
+func (c CommitmentType) String() string {
+	switch c {
+	case OptimismCommitmentType:
+		return "optimism"
+	case NakedCommitmentType:
+		return "naked"
+	default:
+		return "unknown"
+	}
+}
+
+func StrToCommitmentType(s string) CommitmentType {
+	switch s {
+	case "optimism":
+		return OptimismCommitmentType
+	case "naked":
+		return NakedCommitmentType
+	default:
+		return UnknownCommitmentType
+	}
+}
+
+func StringToCommit(key string, c CommitmentType) ([]byte, error) {
+	if len(key) == 0 {
```

Comment:
Should be < 2 since we're indexing on that offset

Reply:
True